### ISSUE TITLE
Wrong localisation of the admin.js script (WP_Scripts::localize)

### ISSUE THREAD

**Issue by timotheemoulin** [polylang/polylang#904]
## Prerequisites

- [x] This is not a usage question
- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [ ] The issue still exists against the latest `master` branch of Polylang and the latest WordPress version.
- [x] This bug happens with only Polylang plugin active
- [x] This bug happens with a default WordPress theme active
- [x] I can reproduce this bug consistently

## Steps to reproduce the issue

1. Install Polylang version 2.9.2
2. Go the an admin page
3. The message is displayed on the page
  > Notice: WP_Scripts::localize est appelée de la mauvaise manière. Le paramètre $l10n doit être un tableau. Pour transmettre des données arbitraires aux scripts, utilisez plutôt la fonction wp_add_inline_script().

Calling `wp_localize_script()` with a string as the 2nd argument has been deprecated.
The issue is producted by `\PLL_Settings::admin_enqueue_scripts` in `settings/settings.php`.

```php
public function admin_enqueue_scripts() {
    parent::admin_enqueue_scripts();

    $suffix = defined( 'SCRIPT_DEBUG' ) && SCRIPT_DEBUG ? '' : '.min';

    wp_enqueue_script( 'pll_admin', plugins_url( '/js/admin' . $suffix . '.js', POLYLANG_FILE ), array( 'jquery', 'wp-ajax-response', 'postbox', 'jquery-ui-selectmenu' ), POLYLANG_VERSION );
    wp_localize_script( 'pll_admin', 'pll_flag_base_url', plugins_url( '/flags/', POLYLANG_FILE ) );
    wp_localize_script( 'pll_admin', 'pll_dismiss_notice', esc_html__( 'Dismiss this notice.', 'polylang' ) );

    wp_enqueue_style( 'pll_selectmenu', plugins_url( '/css/selectmenu' . $suffix . '.css', POLYLANG_FILE ), array(), POLYLANG_VERSION );
}
```

I don't think that `pll_flag_base_url` is used anywhere so it might easily be removed.
`pll_dismiss_notice` is only used by `js/admin.js:309`, but is wrongly used in several places.

**Comment by timotheemoulin** [comment]
The error has been fixed in `master`, can it be ported to the `2.9` branch?
https://github.com/polylang/polylang/blob/master/settings/settings.php#L334

**Comment by Chouby** [comment]
As you noticed yourself, this was already fixed since v3.0 in #710.
Unlike WP core which supports one update per branch, the plugins update mechanism supports only the update to the latest version. If today you are using Polylang 2.9.1, WP proposes you to update to 3.1, not 2.9.2. And to be fully honest, even if updates per branch would be available, we would backport only security fixes.
